tests/widgets/history/HistoryComparisonTest.php

```
@@ -0,0 +1,235 @@
+        );
+        self::assertSame(
+            1,
+            $panel->changed,
+            'Shared paths with different typed values must be counted.',
+        );
+        self::assertSame(
+            2,
+            $panel->unchanged,
+            'Shared paths with equal typed values must be counted.',
+        );
+    }
```

PHPMD (2.15.0)

[warning] The method fromSnapshotsComputesMetricAndPanelDifferencesWithoutExposingValues() has 120 lines of code. Current threshold is set to 100. Avoid really long methods. (undefined)

(ExcessiveMethodLength)